source/postprocess/visualization/fluid_velocity.cc

```diff
@@ -0,0 +1,177 @@
+
+              in.reinit(fe_values, cell, this->introspection(), this->get_solution());
+
+              this->get_material_model().create_additional_named_outputs(out);
+              this->get_material_model().evaluate(in, out);
+              this->get_material_model().fill_additional_material_model_inputs(in, this->get_solution(), fe_values, this->introspection());
+              fluid_out = out.template get_additional_output<MaterialModel::MeltOutputs<dim>>();
+
+              for (unsigned int q=0; q<n_q_points; ++q)
+                {
+                  const unsigned int porosity_idx = this->introspection().find_composition_type(CompositionalFieldDescription::porosity);
+                  const double porosity = std::max(in.composition[q][porosity_idx], 0.0);
+                  const Tensor<1,dim> gravity = this->get_gravity_model().gravity_vector(in.position[q]);
+                  const double solid_density = out.densities[q];
+                  // const double fluid_viscosity = fluid_out->fluid_viscosities[q];
+                  // const double fluid_density = fluid_out->fluid_densities[q];
+                  // const double permeability = fluid_out->permeabilities[q];
+                  const double fluid_viscosity = 10;
+                  const double fluid_density = 1000;
+                  const double permeability = 1e-12;
+                  const Tensor<1,dim> fluid_velocity = solid_velocities[q] -
+                                                       permeability / fluid_viscosity / porosity * gravity *
+                                                       (solid_density - fluid_density) * velocity_scaling_factor;
+
+                  if (dim==2)
+                    {
+                      computed_quantities[q](0) =  fluid_velocity[0];
+                      computed_quantities[q](1) = fluid_velocity[1];
+                    }
+
+                  if (dim==3)
+                    {
+                      computed_quantities[q](0) =  fluid_velocity[0];
+                      computed_quantities[q](1) = fluid_velocity[1];
+                      computed_quantities[q](2) = fluid_velocity[2];
+                    }
+
+                }
+            }
+      }
+
+
+      template <int dim>
+      std::vector<std::string>
+      FluidVelocity<dim>::get_names () const
+      {
+        std::vector<std::string> names;
+        names.emplace_back("fluid_velocity_x");
+        names.emplace_back("fluid_velocity_y");
+        if (dim==3)
+          names.emplace_back("fluid_velocity_z");
+        return names;
+      }
+
+
+      template <int dim>
+      std::vector<DataComponentInterpretation::DataComponentInterpretation>
+      FluidVelocity<dim>::get_data_component_interpretation () const
+      {
+        return
+          std::vector<DataComponentInterpretation::DataComponentInterpretation>
+          (dim,DataComponentInterpretation::component_is_scalar);
```

Comment:
I am not sure exactly, but I think this way you'll get three scalars rather than one vector. Look at how the gravity postprocessor does it, that's way simpler.

Reply:
Youre right! It is the scalar components not a vector